src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -176,7 +177,33 @@ public void startTask(String tableName, TopicPartition topicPartition) {
     createTableIfNotExists(tableName);
 
     // Create channel for the given partition
-    createStreamingChannelForTopicPartition(tableName, topicPartition);
+    createStreamingChannelForTopicPartition(tableName, topicPartition, null);
+  }
+
+  /**
+   * Initializes multiple Channels and partitionsToChannel maps with new instances of {@link
+   * TopicPartitionChannel}
+   *
+   * @param partitions collection of topic partition
+   * @param topic2Table map of topic to table name
+   */
+  @Override
+  public void startTasks(Collection<TopicPartition> partitions, Map<String, String> topic2Table) {
+    Map<String, Boolean> tableName2SchemaEvolutionPermission = new HashMap<>();
+    partitions.forEach(
+        tp -> {
+          String tableName = Utils.tableName(tp.topic(), topic2Table);
+
+          if (!tableName2SchemaEvolutionPermission.containsKey(tableName)) {
+            createTableIfNotExists(tableName);
```

Comment:
Looks like the startTask could be removed now given that we're not using it anymore?

Reply:
Still using it [here](https://github.com/snowflakedb/snowflake-kafka-connector/blob/e63dbfc17ddf0e045a38b9b19ed6295ce34a7877/src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceFactory.java#L70)